### ISSUE TITLE
Unexpected kernel_shape (and trainable parameters) in Keras group convolution

### ISSUE THREAD

**Issue by raulsoutelo** [tensorflow/tensorflow#45751]
In Google Colab (tensorflow version 2.3.0) I am getting an unexpected kernel shape (and trainable params in the layer) for keras group convolution. 

```
import tensorflow as tf
print(tensorflow.__version__)

batch = 15
channels = 6
rows = 10
cols = 10
kernel = (3, 3)
groups = 2
filters = 32

conv_layer = tf.keras.layers.Conv2D(filters, kernel, groups)

output = conv_layer(tf.ones([batch, rows, cols, channels]))

print(conv_layer.get_weights()[0].shape)
print(conv_layer.get_weights()[1].shape)
```

gives me the following output:

```
2.3.0
(3, 3, 6, 32)
(32,)
```

Reading the documentation for the groups params in Conv1D/Conv2D/Conv3D, I was expecting a kernel shape of the following dimensions: weights (3, 3, 6, 16) and biases (16,). Is that a bug or I am missing something?

Thanks!

**Comment by ravikyram** [comment]
I have tried in colab with TF version 2.2, 2.3 and nightly version(`2.5.0-dev20201216`) and was able to reproduce the issue. Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/191e8f4f718524085cdad0f5ea139be8/untitled581.ipynb). Thanks!

**Comment by raulsoutelo** [comment]
Sorry, my previous code snippet had a bug since groups is not the third param in Conv2D. Here is an updated version:

```
import tensorflow as tf
print(tf.__version__)

batch =15
channels=6
rows = 10
cols = 10
kernel = (3, 3)
groups = 2
filters = 32

conv_layer = tf.keras.layers.Conv2D(filters, kernel, groups=groups)

output = conv_layer(tf.ones([batch, rows, cols, channels]))

print(conv_layer.get_weights()[0].shape)
print(conv_layer.get_weights()[1].shape)
```

This prints

```
2.4.0
(3, 3, 3, 32)
(32,)
```

which corresponds to: kernel + channels/2 + filters. I was expecting kernel + channels/2 + filters/2.

I have added here and additional code snippet that in my view seems to contradict

> groups: A positive integer specifying the number of groups in which the input is split along the channel axis. Each group is convolved separately with filters / groups filters. The output is the concatenation of all the groups results along the channel axis. Input channels and filters must both be divisible by groups.

```
import tensorflow as tf
print(tf.__version__)

batch =1
channels=4
rows = 4
cols = 4
kernel = 3
groups = 2
filters = 6

# create two random tensors. Each of them will be a group
random_input1 = tf.random.uniform(shape=(batch, rows, cols, channels//2))
random_input2 = tf.random.uniform(shape=(batch, rows, cols, channels//2))

# create a conv layer with param groups equal to 2
conv_layer = tf.keras.layers.Conv2D(filters, kernel, groups=groups)

# perform convolution operation twice (groups concatenated in different order)
# the first (filters/2) filters in the output should correspond to the first 
# group provided and the last (filters/2) filters to the second one
output1 = conv_layer(tf.keras.backend.concatenate([random_input1, random_input2], axis=3))
output2 = conv_layer(tf.keras.backend.concatenate([random_input2, random_input1], axis=3))

# extract first and second half (along channels axis) for output2. First half
# should correspond to first group (random_input2) and second half to random_input1
output2_first = tf.slice(output2, [0, 0, 0, 0], [output2.shape[0], output2.shape[1], output2.shape[2], output2.shape[3]//2])
output2_second = tf.slice(output2, [0, 0, 0, output2.shape[3]//2], [output2.shape[0], output2.shape[1], output2.shape[2], output2.shape[3]//2])

# Create a new tensor where filters corresponding to random_input1 go first
# This should be equivalent to output1 where group1 was random_input1 and group2 was random_input2
output2_transformed = tf.keras.backend.concatenate([output2_second, output2_first], axis=3)

# Print difference between two tensors. It should be 0.
diff = output1 - output2_transformed
print(diff)
```

This print a tensor with values different to zero.

**Comment by rmothukuru** [comment]
@raulsoutelo,
The behavior is expected because, as per the [Documentation of Groups Argument](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2D#arguments_1), 

> The output is the concatenation of all the groups results along the channel axis.

and as per [this blog](https://blog.yani.ai/filter-group-tutorial/) , 

> each filter group are convolved with only half the previous layer’s featuremaps

From the above statements, **`Feature Maps`** from previous layers are divided among 2 **`Groups`** but finally, they are concatenated. 

Hence, the **`Number of Parameters`** with and without **`Groups`** (by the end of the Layer) will be same but the Representational Power is better.

**Comment by rmothukuru** [comment]
@raulsoutelo,
Can you please respond to the [above comment](https://github.com/tensorflow/tensorflow/issues/45751#issuecomment-750035881)? Thanks!

**Comment by raulsoutelo** [comment]
Thanks for the clarification @rmothukuru. I initially thought this group parameter was the size of the group in Group Equivariant Convolutional Networks which is not the case. The previous comment is clear to me. Thanks!
